### DISCUSSION TITLE
Slicer Button + Bookmarks

### DISCUSSION THREAD

**Discussion by MattNgt** [mattkocak/powerbi-visuals-slicerbutton#4] [Q&A]
Hello Matt,
You have created a great visual, thanks for that, it is very useful.
I was wondering if it is possible to combine the slicer button with bookmarks ?
To explain my need : I use the slicer button to filter data on a table to show certain values for a column, but I have a slicer buttons for each calculated columns that I have. I would like the table to be dynamic : like when I click on the slicer button associated to column A I would like to have in my table two columns (ID, Column A), when i click on the slicer button for column B  I want a table with ID and Column B.

With the use of bookmarks I can make one bookmark for each column to show only the columns that I want in my table, but that don't combine with the slicer. 
I understand if that's not possible but it could be a great feature to your visual.

Thanks again :)

Matt

**Comment by mattkocak** [comment]
Hey Matt,

Thanks for reaching out. I understand your request, but would like to know some more specifics to see the best way to address this. Would you please provide some sample data for 3 columns (Id, Column A, Column B) in [table format](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables) as well as the slicer settings/bookmarks and expected results? I've included a template below that you can use.

**Sample Data**
| Id | Column A | Column B |
| --- | --- | --- |
| ... | ... | ... |
| ... | ... | ... |
| ... | ... | ... |
| ... | ... | ... |
| ... | ... | ... |
| ... | ... | ... |

**Slicer Button A:** _<ex. Column: A, Values: 1,2>_
**Bookmark A:** _<ex. show Id and Column A>_
**Result A:**
| Id | Column A |
| --- | --- |
| ... | ... |
| ... | ... |

**Slicer Button A:** _<ex. Column: B, Values: 3,4>_
**Bookmark A:** _<ex. show Id and Column B>_
**Result B:**
| Id | Column B |
| --- | --- |
| ... | ... |
| ... | ... |

Thanks,
Matt

**Reply by MattNgt** [reply]
Hey Matt,

Thanks for your quick answer. Yes, it would be clearer :

<html><body>


Id | Column A | Column B
-- | -- | --
B | 1 | 1
C | 0 | 1
D | 1 | 0
E | 0 | 0
F | 1 | 0
G | 1 | 1


</body>
</html>

Slicer Button A: <ex. Column: A, Values: 1>
Bookmark A: <ex. show Id and Column A>

<html><body>
<p dir="auto"><strong>Result A:</strong></p>


Id | Column A 
-- | -- 
B | 1 
D | 1
F | 1
G | 1

</body>
</html>

Slicer Button B: <ex. Column: B, Values: 1>
Bookmark B: <ex. show Id and Column B>
Result B:

<html><body>


Id | Column B
-- | --
B | 1
C | 1
G | 1

</body>
</html>

Hope it's better like that !

Thanks, 
Matt

**Reply by mattkocak** [reply]
Thanks for the info. Unfortunately, I won't be implementing a bookmark feature as it adds a lot of complexity in managing the interactions between slicers and bookmarks.

Could the problem you are facing be solved strictly with bookmarks? I'm assuming you're using bookmarks to show/hide different tables that display different columns, giving the effect that it's the same table behaving dynamically. Could you also use these bookmarks to filter the tables with the data that you want to see?

The [Understand Power BI Filters](https://data-flair.training/blogs/power-bi-filter/#:~:text=Types%20of%20Filters%20in%20Power%20BI) article details setting filters at various levels of a report. Also, there's the ability to only apply filter/display changes to specifically selected visuals as outlined in the [Advanced Bookmarks](https://www.designmind.com/blog/business-intelligence/bookmarks-and-buttons-in-power-bi#:~:text=Updating%20Bookmark%20Using%20%E2%80%98Selected%20Visuals%E2%80%99) article. This combination of concepts can make for some pretty diverse functionality that may accomplish what you're looking to achieve.

Anyway, hope this helps and that you find a solution that suits you. Thanks for reaching out on the visual and let me know you have any additional input.

Best,
Matt

**Comment by mattkocak** [comment]
Closing as not implementing.
